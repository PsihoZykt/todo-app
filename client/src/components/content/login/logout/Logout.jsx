import React from 'react';
import { withRouter } from 'react-router-dom';

const Logout = (props) => {
  let onLogout = () => {
    props.logout()

  }
  return(

    <span tabIndex={0} role="button" onClick={onLogout}>
    Logout
  </span>
  );
}

export default withRouter(Logout);
