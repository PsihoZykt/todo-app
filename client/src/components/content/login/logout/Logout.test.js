import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Logout from './index';

describe('Logout', () => {
  it('should render correctly', () => {
    const output = shallow(<Logout logout={() => 'simulated logout'} />);
    expect(shallowToJson(output)).toMatchSnapshot();
    output.find('[role="button"]').simulate('click');
    expect(output).toMatchSnapshot();
  });
});
