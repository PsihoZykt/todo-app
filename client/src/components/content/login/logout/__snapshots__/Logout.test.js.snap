// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;

exports[`Link should render correctly 2`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;

exports[`Logout should render correctly 1`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;

exports[`Logout should render correctly 2`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;

exports[`test onClick handler 1`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;

exports[`test onClicl handler 1`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;

exports[`test onClicl handler 2`] = `
<span
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  Logout
</span>
`;
